test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeTests.java

```diff
@@ -399,6 +455,1338 @@ static public void testWithFieldOnNonExistentClass() throws Throwable {
+		Object valueFloat = makeValueFloat.invoke(f);
+
+		assertEquals(getFloat.invoke(valueFloat), f);
+
+		valueFloat = withFloat.invoke(valueFloat, fNew);
+		assertEquals(getFloat.invoke(valueFloat), fNew);
+	}
+
+	/*
+	 * Create a value type with a reference member
+	 * 
+	 * value ValueObject {
+	 * 	Object val;
+	 * }
+	 */
+	@Test(priority = 1)
+	static public void testCreateValueObject() throws Throwable {
+		String fields[] = { "val:Ljava/lang/Object;:value" };
+
+		valueObjectClass = ValueTypeGenerator.generateValueClass("ValueObject", fields);
+
+		makeValueObject = lookup.findStatic(valueObjectClass, "makeValue",
+				MethodType.methodType(valueObjectClass, Object.class));
+
+		Object val = (Object)0xEEFFEEFF;
+		Object valNew = (Object)0xFFEEFFEE;
+
+		getObject = generateGetter(valueObjectClass, "val", Object.class);
+		withObject = generateWither(valueObjectClass, "val", Object.class);
+
+		Object valueObject = makeValueObject.invoke(val);
+
+		assertEquals(getObject.invoke(valueObject), val);
+
+		valueObject = withObject.invoke(valueObject, valNew);
+		assertEquals(getObject.invoke(valueObject), valNew);
+	}
+
+	/*
+	 * Create an assorted value type with long alignment
+	 * 
+	 * value AssortedValueWithLongAlignment {
+	 *	flattened Point2D point;
+	 *	flattened Line2D line;
+	 *	flattened ValueObject o;
+	 *	flattened ValueLong l;
+	 *	flattened ValueDouble d;
+	 *	flattened ValueInt i;
+	 *	flattened Triangle2D tri;
+	 * }
+	 */
+	@Test(priority = 4)
+	static public void testCreateAssortedValueWithLongAlignment() throws Throwable {
+		String fields[] = {
+				"point:QPoint2D;:value",
+				"line:QFlattenedLine2D;:value",
+				"o:QValueObject;:value",
+				"l:QValueLong;:value",
+				"d:QValueDouble;:value",
+				"i:QValueInt;:value",
+				"tri:QTriangle2D;:value" };
+		Class assortedValueWithLongAlignmentClass = ValueTypeGenerator
+				.generateValueClass("AssortedValueWithLongAlignment", fields);
+
+		MethodHandle makeAssortedValueWithLongAlignment = lookup.findStatic(assortedValueWithLongAlignmentClass,
+				"makeValueGeneric", MethodType.methodType(assortedValueWithLongAlignmentClass, Object.class,
+						Object.class, Object.class, Object.class, Object.class, Object.class, Object.class));
+
+		/*
+		 * Getters are created in array getterAndWither[i][0] according to the order of fields i
+		 * Withers are created in array getterAndWither[i][1] according to the order of fields i
+		 */
+		MethodHandle[][] getterAndWither = generateGenericGetterAndWither(assortedValueWithLongAlignmentClass, fields);
+
+		/* point */
+		int[] pointFields = { 0xFFEEFFEE, 0xAABBAABB };
+		Object point = makePoint2D.invoke(pointFields[0], pointFields[1]);
+		checkEqualPoint2D(point, pointFields);
+
+		/* line */
+		int[][] lineFields = { { 0xFFEEFFEE, 0xAABBAABB }, { 0xCCAFFAAA, 0xBBAAAACC } };
+		Object line = createFlattenedLine2D(lineFields);
+		checkEqualLine2D(line, lineFields, true);
+
+		/* triangle */
+		int[][][] triangleFields = {
```

Comment:
can you make a helper for the large objects (triangle,  AssortedValueWith..., etc) so there isnt som much code duplication

Reply:
There are some helper functions like `createTriangle2D(int[] triangleFields)`.
do you want a default construction function(i.e a creation function without providing any fields)?

Do we need helper functions for assorted valuetypes? Since they are objects at the top, there are not too much code duplication